Remove crypto-only related components from all.sh

AndrzejKurek opened the pull request:
This PR handles the removal of crypto-only components from all.sh tests.
More details [here](https://jira.arm.com/browse/IOTSSL-2800).

gilles-peskine-arm commented:
@RonEld 

* `component_test_depends_hashes`, `component_test_depends_pkalgs`: these check dependencies in X.509 tests and `#ifdef` guards in X.509 and TLS code.
* `component_test_null_entropy`: Although the corresponding `#ifdef` is only used in a crypto module, this could have an impact on how some programs run, so I think it should be tested in a broader context.
* `component_test_min_mpi_window_size`: ah, that's new. It wasn't there when we looked at what to remove. I agree that it should be removed. Thanks for spotting it!
* `component_test_no_udbl_division`, `component_test_no_64bit_multiplication`, `component_build_arm_none_eabi_gcc_no_udbl_division`, `component_build_arm_none_eabi_gcc_no_64bit_multiplication`: these are borderline. In principle they apply to the whole source base, but in practice only some crypto code is likely to violate them. My recommendation is to keep the `build_arm_` ones and remove the `test_` ones because these test different things:
    * If the corresponding `#ifdef` breaks something, it should be in the crypto code and caught by crypto's corresponding `test_` components.
    * If there's code that uses a forbidden operation, it could be anywhere, and that's what the `build_arm_` components test.

RonEld commented:
Thanks @gilles-peskine-arm for your answer.
As long as the sections I mentioned were *knowingly* kept, I'm fine with that.
I'll approve once `component_test_min_mpi_window_size`, `component_test_no_udbl_division` and `component_test_no_64bit_multiplication` will be removed, according to your recomendation

AndrzejKurek commented:
@RonEld - as for the `component_test_min_mpi_window_size` - my bad, I haven't noticed it after rebasing. As for the rest of the build_arm ones - I wasn't  sure about these, so thanks for your input @gilles-peskine-arm  and @RonEld.
I have now amended the previous commit to contain all changes.

Patater commented:
Please resolve conflicts (by deciding what to do about `component_test_new_ecdh_context`). Thanks.

AndrzejKurek commented:
I have rebased this PR and removed the forementioned component - `component_test_new_ecdh_context`.
@RonEld, @gilles-peskine-arm  - please re-review.

Patater commented:
CI failure on head is flaky DTLS test. The merge result tests fine, so ready to merge.
